File: internal/score/mirrors_test.go
package score

import "testing"

func TestFeedMirrorsFor(t *testing.T) {
	got := FeedMirrorsFor("https://css-tricks.com/feed/")
	if len(got) != 1 || got[0] != "https://feeds.feedburner.com/CssTricks" {
		t.Fatalf("css-tricks mirrors: %v", got)
	}
	got = FeedMirrorsFor("https://www.css-tricks.com/feed/")
	if len(got) != 1 {
		t.Fatalf("www css-tricks mirrors: %v", got)
	}
	if FeedMirrorsFor("https://example.com/feed") != nil {
		t.Fatal("expected no mirrors for example.com")
	}
}

File: workers/scan/src/mirrors.test.ts
import { describe, expect, it } from 'vitest'
import { feedMirrorsFor } from './mirrors'

describe('feedMirrorsFor', () => {
  it('returns Feedburner for css-tricks.com feed URLs', () => {
    expect(feedMirrorsFor('https://css-tricks.com/feed/')).toEqual([
      'https://feeds.feedburner.com/CssTricks',
    ])
    expect(feedMirrorsFor('https://www.css-tricks.com/feed/')).toEqual([
      'https://feeds.feedburner.com/CssTricks',
    ])
  })

  it('returns empty for unknown hosts', () => {
    expect(feedMirrorsFor('https://example.com/feed')).toEqual([])
  })
})
